lib/output.js

```diff
@@ -1964,19 +1964,14 @@ function OutputStream(options) {
     function make_num(num) {
         var str = num.toString(10), a = [ str.replace(/^0\./, ".").replace("e+", "e") ], m;
         if (Math.floor(num) === num) {
-            if (num >= 0) {
-                a.push("0x" + num.toString(16).toLowerCase(), // probably pointless
-                       "0" + num.toString(8)); // same.
-            } else {
-                a.push("-0x" + (-num).toString(16).toLowerCase(), // probably pointless
-                       "-0" + (-num).toString(8)); // same.
-            }
-            if ((m = /^(.*?)(0+)$/.exec(num))) {
+            a.push(
+                (num >= 0 ? "0x" : "-0x") + num.toString(16).toLowerCase() // probably pointless
+            );
+            if ((m = /^(.*?)(0{3,})$/.exec(num))) {
```

Comment:
If I understand correctly, changing `(0+)` to `(0{3,})` is just to avoid considering `1e1` and `1e2`, which are always shorter or equally long as their decimal equivalent. Right?

Reply:
Yea, 10 is shorter than 1e1

Reply:
Yea, 10 is shorter than 1e1